docs/conf.py

```diff
@@ -156,7 +156,22 @@ def check_dot_command():
 # List of patterns, relative to source directory, that match files and
 # directories to ignore when looking for source files.
 # The documentation build tool overrides this when extensions are included in the documentation build.
-exclude_patterns = ['adr/*.md']
+
+# Make sure iso-only documents are excluded from the doc build by default and only included for iso builds.
+# We can't use the  `.. only:: iso` directive here because it is only suited to control the content of
+# the documents and not their structure.
+
+
+default_pattern = ['adr/*.md']
+exclude_from_oss = [
+    "lsm",
+    "administrators/operational_procedures_with_lsm.rst",
+    "administrators/support.rst"
+]
+exclude_patterns = default_pattern + exclude_from_oss
+
+if tags.has("iso"):
+    exclude_patterns = default_pattern
```

Comment:
This is a very strange pattern imo:
```
default = ...
x = default + ...  # add something to it
if c:
  x = default  # changed our minds
```

Instead of this back-and-forth, wouldn't it be a lot more straightforward to either define it at once (functional, my preference), or at least to only move in one direction (not add first, then redefine: only add).

In addition to this, the behavior is not exactly the same as with the `only` directives. Perhaps we want to condition on `not tags.has("iso")` and `not tags.has("oss")` instead.